diff --git a/configure.py b/configure2.py
index f7392d0..ecae4bb 100644
--- a/configure.py
+++ b/configure2.py
@@ -9,6 +9,7 @@ import os
 import subprocess
 import sys
 import textwrap
+import re
 
 
 base_dir = os.path.abspath(os.path.dirname(__file__))
@@ -56,6 +57,14 @@ def config_status(config):
     sanitized_config['topobjdir'] = config['TOPOBJDIR']
     sanitized_config['mozconfig'] = config.get('MOZCONFIG')
 
+    # Change python _virtualenv path to use a relative one. 
+    # Using Yocto the shebang limit was exceeded.
+    paths = [config['TOPOBJDIR'],config['PYTHON']]
+    common_prefix = os.path.commonprefix(paths)
+    relative_path=os.path.relpath(config['TOPOBJDIR'], common_prefix)
+    if relative_path != '.':
+        relative_path=re.sub("[ a-zA-Z0-9\._-]+","..",relative_path)
+    
     # Create config.status. Eventually, we'll want to just do the work it does
     # here, when we're able to skip configure tests/use cached results/not rely
     # on autoconf.
@@ -63,7 +72,7 @@ def config_status(config):
     encoding = 'mbcs' if sys.platform == 'win32' else 'utf-8'
     with codecs.open('config.status', 'w', encoding) as fh:
         fh.write(textwrap.dedent('''\
-            #!%(python)s
+            #!'''+relative_path+'''/_virtualenv/bin/python
             # coding=%(encoding)s
             from __future__ import unicode_literals
             from mozbuild.util import encode
