--- firefox-52.0b8/gfx/2d/HelpersCairo.h	2017-03-02 09:23:01.230766731 -0500
+++ firefox-52.0b8/gfx/2d/HelpersCairo.h	2017-03-02 09:23:49.711930251 -0500
@@ -239,9 +239,9 @@ CairoFormatToGfxFormat(cairo_format_t fo
 {
   switch (format) {
     case CAIRO_FORMAT_ARGB32:
-      return SurfaceFormat::A8R8G8B8_UINT32;
+      return SurfaceFormat::B8G8R8A8;
     case CAIRO_FORMAT_RGB24:
-      return SurfaceFormat::X8R8G8B8_UINT32;
+      return SurfaceFormat::B8G8R8X8;
     case CAIRO_FORMAT_A8:
       return SurfaceFormat::A8;
     case CAIRO_FORMAT_RGB16_565:
