diff --git a/forms/source/xforms/datatypes.hxx b/forms/source/xforms/datatypes.hxx
index ee00452e5..c5ef154ba 100644
--- a/forms/source/xforms/datatypes.hxx
+++ b/forms/source/xforms/datatypes.hxx
@@ -36,6 +36,8 @@
 
 #include <memory>
 
+using icu::RegexMatcher;
+
 
 namespace xforms
 {
diff --git a/i18npool/inc/collator_unicode.hxx b/i18npool/inc/collator_unicode.hxx
index 69d317797..2b4c2b9bf 100644
--- a/i18npool/inc/collator_unicode.hxx
+++ b/i18npool/inc/collator_unicode.hxx
@@ -26,6 +26,8 @@
 
 #include <unicode/tblcoll.h>
 
+using icu::RuleBasedCollator;
+
 //      ----------------------------------------------------
 //      class Collator_Unicode
 //      ----------------------------------------------------
diff --git a/i18npool/source/breakiterator/breakiterator_unicode.cxx b/i18npool/source/breakiterator/breakiterator_unicode.cxx
index b4ca11117..7a7fb4f35 100644
--- a/i18npool/source/breakiterator/breakiterator_unicode.cxx
+++ b/i18npool/source/breakiterator/breakiterator_unicode.cxx
@@ -30,6 +30,9 @@
 #include <rtl/ustring.hxx>
 #include <string.h>
 
+using icu::BreakIterator;
+using icu::RuleBasedBreakIterator;
+
 U_CDECL_BEGIN
 extern const char OpenOffice_dat[];
 U_CDECL_END
diff --git a/i18npool/source/collator/collator_unicode.cxx b/i18npool/source/collator/collator_unicode.cxx
index 22b7ed9ae..08ac8f71a 100644
--- a/i18npool/source/collator/collator_unicode.cxx
+++ b/i18npool/source/collator/collator_unicode.cxx
@@ -29,6 +29,8 @@
 #include <com/sun/star/i18n/CollatorOptions.hpp>
 #include <cppuhelper/supportsservice.hxx>
 
+using icu::Collator;
+
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::i18n;
 using namespace ::com::sun::star::lang;
diff --git a/i18npool/source/collator/gencoll_rule.cxx b/i18npool/source/collator/gencoll_rule.cxx
index 7d795b5af..6d70106dd 100644
--- a/i18npool/source/collator/gencoll_rule.cxx
+++ b/i18npool/source/collator/gencoll_rule.cxx
@@ -31,6 +31,8 @@
 
 #include <unicode/tblcoll.h>
 
+using icu::RuleBasedCollator;
+
 /* Main Procedure */
 
 void data_write(char* file, char* name, sal_uInt8 *data, sal_Int32 len)
diff --git a/i18npool/source/ordinalsuffix/ordinalsuffix.cxx b/i18npool/source/ordinalsuffix/ordinalsuffix.cxx
index a3e7d4ab6..b819d1859 100644
--- a/i18npool/source/ordinalsuffix/ordinalsuffix.cxx
+++ b/i18npool/source/ordinalsuffix/ordinalsuffix.cxx
@@ -29,6 +29,8 @@
 #include <unicode/normlzr.h>
 #include <memory>
 
+using icu::NumberFormat;
+
 using namespace ::com::sun::star::i18n;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star;
diff -u a/i18npool/source/transliteration/ignoreDiacritics_CTL.cxx b/i18npool/source/transliteration/ignoreDiacritics_CTL.cxx 
--- a/i18npool/source/transliteration/ignoreDiacritics_CTL.cxx    2016-05-04 23:28:24.000000000 +0200
+++ b/i18npool/source/transliteration/ignoreDiacritics_CTL.cxx  2018-08-20 18:55:03.599032281 +0200                                                                                                                                                                                                                                                           
@@ -10,6 +10,10 @@                                                                                                                                                                                                                                                             
 #include <transliteration_Ignore.hxx>
 #include <unicode/uchar.h>
 
+#include <unicode/unistr.h>
+
+using icu::UnicodeString;
+
 namespace com { namespace sun { namespace star { namespace i18n {
 
 sal_Unicode
diff --git a/i18nutil/source/utility/unicode.cxx b/i18nutil/source/utility/unicode.cxx
index 0173a4ad5..e937c070e 100644
--- a/i18nutil/source/utility/unicode.cxx
+++ b/i18nutil/source/utility/unicode.cxx
@@ -34,6 +34,9 @@
 // which (obviously) breaks UnicodeType::CURRENCY_SYMBOL
 #undef CURRENCY_SYMBOL
 
+using icu::NumberFormat;
+using icu::UnicodeString;
+
 using namespace ::com::sun::star::i18n;
 
 template<class L, typename T>
diff -u a/lotuswordpro/source/filter/localtime.cxx b/lotuswordpro/source/filter/localtime.cxx                                                                                                    
--- a/lotuswordpro/source/filter/localtime.cxx    2016-05-04 23:28:24.000000000 +0200                                                                                                             
+++ b/lotuswordpro/source/filter/localtime.cxx  2018-08-20 18:46:07.485009107 +0200                                                                                                             
@@ -56,6 +56,9 @@                                                                                                                                                                               
 #include "localtime.hxx"
 #include <limits.h>
 #include <unicode/timezone.h>
+
+using icu::TimeZone;
+
 //End by
 const long DAY_SEC =24 * 60 * 60;
 const long YEAR_SEC = 365 * DAY_SEC;
diff --git a/opencl/source/openclconfig.cxx b/opencl/source/openclconfig.cxx
index b81ffed22..e7b2523f9 100644
--- a/opencl/source/openclconfig.cxx
+++ b/opencl/source/openclconfig.cxx
@@ -17,6 +17,8 @@
 #include <sal/log.hxx>
 #include <sal/types.h>
 
+using icu::RegexMatcher;
+
 OpenCLConfig::OpenCLConfig() :
     mbUseOpenCL(true)
 {
diff -u a/vcl/generic/glyphs/scrptrun.h b/vcl/generic/glyphs/scrptrun.h 
--- a/vcl/generic/glyphs/scrptrun.h       2016-05-04 23:28:24.000000000 +0200
+++ b/vcl/generic/glyphs/scrptrun.h     2018-08-20 19:13:50.475065945 +0200
@@ -45,6 +45,8 @@
 #include "unicode/uscript.h"
 #include <vector>
 
+using icu::UObject;
+
 namespace vcl {
 
 struct ScriptRecord
