diff --git a/configure.ac.orig b/configure.ac
index 97ffe85..6e42a7e 100644
--- a/configure.ac.orig
+++ b/configure.ac
@@ -78,12 +78,9 @@ AC_CHECK_FUNCS([getpeerucred getpeereid memset setenv strchr strdup \
                 strerror strrchr strspn strstr strtol strtoul uname \
                 setlocale mount umount unmount])
 
-AC_CHECK_MEMBERS([struct stat.st_rdev])
-
-AC_CHECK_FILE(/sys/class/tty/tty0/active,
-              [AC_DEFINE([HAVE_SYS_VT_SIGNAL], [1], [System has a means of signaling VT changes])],
-              [])
+AC_CHECK_MEMBERS([struct stat.st_rdev])
 
+AC_DEFINE([HAVE_SYS_VT_SIGNAL], [1], [System has a means of signaling VT changes])
 
 XDT_CHECK_PACKAGE([LIBDBUS], [dbus-1], [dbus_minimum_version])
 XDT_CHECK_PACKAGE([GLIB], [glib-2.0], [glib_minimum_version])
